ClosedXML/ClosedXML.csproj

```diff
@@ -1,386 +1,53 @@
+
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>8.0.30703</ProductVersion>
-    <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{BD5E6BFE-E837-4A35-BCA9-39667D873A20}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>ClosedXML</RootNamespace>
-    <AssemblyName>ClosedXML</AssemblyName>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
-    <RestorePackages>true</RestorePackages>
-    <TargetFrameworkProfile />
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
-    <Optimize>false</Optimize>
-    <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <NoWarn>1591</NoWarn>
-    <Prefer32Bit>false</Prefer32Bit>
-    <LangVersion>6</LangVersion>
+    <TargetFrameworks>netstandard2.0;net40;net46</TargetFrameworks>
+    <Version>0.9</Version>
+    <Authors>Manuel de Leon, Amir Ghezelbash, Francois Botha</Authors>
+    <Company />
+    <Product />
+    <Description>ClosedXML makes it easier for developers to create Excel 2007+ (.xlsx, .xlsm, etc) files. It provides a nice object oriented way to manipulate the files (similar to VBA) without dealing with the hassles of XML Documents. It can be used by any .NET language like C# and VisualBasic.NET.</Description>
+    <Copyright>MIT</Copyright>
+    <PackageLicenseUrl>https://github.com/ClosedXML/ClosedXML/blob/master/LICENSE</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/ClosedXML/ClosedXML</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/ClosedXML/ClosedXML</RepositoryUrl>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>ClosedXML.snk</AssemblyOriginatorKeyFile>
+    <NoWarn>$(NoWarn);NU1605;CS1591</NoWarn>
+    <DocumentationFile>$(PublishDir)\ClosedXML.xml</DocumentationFile>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <DocumentationFile>bin\Release\ClosedXML.xml</DocumentationFile>
-    <NoWarn>1591</NoWarn>
-    <Prefer32Bit>false</Prefer32Bit>
-    <LangVersion>6</LangVersion>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <DefineConstants>$(DefineConstants);_NETSTANDARD_;_NETSTANDARD2_0_</DefineConstants>
```

Comment:
I noticed an inconsistency in defined constants. Here we have ```'$(TargetFramework)' == 'netstandard2.0'``` and ```_NETSTANDARD2_0_```, in Examples and Sandbox - ```'$(TargetFramework)' == 'netcoreapp2.0'``` and ```_NETSTANDARD2_0_``` as well, and in Tests - ```'$(TargetFramework)' == 'netcoreapp2.0'``` and ```_NETCOREAPP2_0_```

Reply:
Yes, that's deliberate. The `ClosedXML` project is a library and therefor I could base it on .NET Standard 2.0. But for the other projects, that are _executable_, you need to specify a target that is actually runnable, not just a standard. That's how I understand it. I could be wrong. We definitely want the `ClosedXML` to be `netstandard2.0`. I'll try to see if i can get the other projects to build and execute on `netstandard2.0` instead of `netcoreapp2.0`, but IIRC, I couldn't.

Reply:
My concerns were about constant names. What directive should one use for code being compiled only in netstandard version? In three projects this would be ```#if _NETSTANDARD2_0_ ...``` while in tests - ```#if _NETCOREAPP2_0_ ...```. I suspected there must be either the same one for all projects or both (for all projects as well).

Reply:
Tried building `ClosedXML_Examples` with `netstandard2.0`. It builds fine, but when I try to run it:

```
c:\dev\ClosedXML\ClosedXML\ClosedXML_Examples>dotnet run -f netstandard2.0
Microsoft (R) Build Engine version 15.6.82.30579 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 39.19 ms for c:\dev\ClosedXML\ClosedXML\ClosedXML\ClosedXML.csproj.
  Restore completed in 39.19 ms for c:\dev\ClosedXML\ClosedXML\ClosedXML_Examples\ClosedXML_Examples.csproj.
Unable to run your project.
Please ensure you have a runnable project type and ensure 'dotnet run' supports this project.
A runnable project should target a runnable TFM (for instance, netcoreapp2.0) and have OutputType 'Exe'.
The current OutputType is 'Exe'.                                  
```

Reply:
Oh I see. It's not a requirement for successful compilation, but I agree it would make it simpler for developers to understand. Will align them tomorrow.

Reply:
They are now all aligned.
` <DefineConstants>$(DefineConstants);_NETSTANDARD_;_NETSTANDARD2_0_</DefineConstants>`